libkbfs/file_data.go

```diff
@@ -290,16 +291,29 @@ func (fd *fileData) getLeafBlocksForOffsetRange(ctx context.Context,
 		respCh := make(chan resp, 1)
 		respChans = append(respChans, respCh)
 		eg.Go(func() error {
-			block, _, err := fd.getter(
-				groupCtx, fd.kmd, ptr, fd.file, blockReadParallel)
-			if err != nil {
-				return err
-			}
-			// Recurse down to the level of the child.
-			pfr, blocks, nextBlockOffset, err := fd.getLeafBlocksForOffsetRange(
-				groupCtx, ptr, block, startOff, endOff, prefixOk)
-			if err != nil {
-				return err
+			var pfr [][]parentBlockAndChildIndex
+			var blocks map[BlockPointer]*FileBlock
+			var nextBlockOffset int64
+			// We only need to fetch direct blocks if we've been asked
+			// to do so.
+			if getDirect || !ptr.IsDirect {
```

Comment:
if we had a tri-state, we could skip this if `getDirect` and `ptr.targetType == Indirect`